lib/mayaUsd/render/vp2ShaderFragments/shaderFragments.cpp

```diff
@@ -237,20 +239,18 @@ MStatus HdVP2ShaderFragments::registerFragments()
 // Fragment deregistration
 MStatus HdVP2ShaderFragments::deregisterFragments()
 {
-    // Similar to registration, we do not force the renderer to initialize in
-    // case we're running in a headless context. If we cannot get a handle to
-    // the renderer or the fragment manager, we assume that's the case and
-    // simply return success.
-    MHWRender::MRenderer* theRenderer =
-        MHWRender::MRenderer::theRenderer(/* initializeRenderer = */ false);
-    if (!theRenderer) {
+    if (!_registered)
         return MS::kSuccess;
+
+    MHWRender::MRenderer* theRenderer = MHWRender::MRenderer::theRenderer();
+    if (!theRenderer) {
+        return MS::kFailure;
     }
 
     MHWRender::MFragmentManager* fragmentManager =
         theRenderer->getFragmentManager();
     if (!fragmentManager) {
-        return MS::kSuccess;
+        return MS::kFailure;
     }
 
     // De-register all fragment graphs.
```

Comment:
Does `_registered` need to be set back to `false` somewhere at the bottom of this function (or in any of the error conditions?) in case the plugin is loaded, unloaded, and then reloaded again?